### PR TITLE
Fix mmday unit error

### LINKED ISSUE #1061: Unit conversions: mm/day and mmday
xclim version: 0.34.0
python version: 3.9.7
Windows 10 

Hello! I believe there is a small bug in unit conversions. If "mmday" is equivalent to "mm/day", then the results are off by a factor of 1000. 

Example:
```
xc.core.units.convert_units_to('1000 mm/day', 'mmday')
#0.9999999999999998
```
It is likely related to this line:
https://github.com/Ouranosinc/xclim/blob/6aba3923282b9c17da986706fbf9ea9b5e303fd2/xclim/core/units.py#L70

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1061
- [x] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [x] `bumpversion patch` has been called on this branch
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* Fix unit definition error for mmday. 

### Does this PR introduce a breaking change?
Bug fix.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
Don't forget to bump version SVP!

File: xclim/testing/tests/test_units.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def test_hydro(self):
         with units.context("hydro"):
             q = 1 * units.kg / units.m**2 / units.s
             assert q.to("mm/day") == q.to("mm/d")
+            assert q.to("mm/day").magnitude == 24 * 60**2
```
Comment (aulemahal) [inline xclim/testing/tests/test_units.py:31]: Did you mean to put: 
```suggestion
            assert q.to("mmday").magnitude == 24 * 60**2
```
?

**Review by aulemahal — APPROVED** [review]
I believe the test you added has a mistake? Otherwise, it's all good.

File: xclim/testing/tests/test_units.py (hunk 1 above)
Comment (huard) [reply xclim/testing/tests/test_units.py:31]: ouaip ! Merci, un peu zoné aujourd'hui...

**Pull request merged**